### PR TITLE
chore(bot): add adaptive card notification sample

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Detail: [User Story 13790838](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/13790838): [Sample] Adaptive Cards

### CODE REVIEW

File: adaptive-card-notification/.fx/configs/projectSettings.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+    "appName": "AdaptiveCardNotification",
+    "projectId": "dac6ca6b-e075-4ebc-9343-88b4c2f047e7",
+    "version": "2.1.0",
+    "isFromSample": false,
```
Comment (a1exwang) [inline adaptive-card-notification/.fx/configs/projectSettings.json:5]: `"isFromSample": true ?

File: adaptive-card-notification/bot/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  "scripts": {
+    "dev:teamsfx": "env-cmd --silent -f .env.teamsfx.local npm run dev",
+    "dev": "func start --typescript --language-worker=\"--inspect=9239\" --port \"3978\" --cors \"*\"",
+    "prepare-storage:teamsfx": "azurite --silent --location ./_storage_emulator --debug ./_storage_emulator/debug.log",
+    "watch:teamsfx": "tsc --watch",
+    "build": "tsc",
+    "watch": "tsc -w",
+    "prestart": "npm run build",
+    "start": "npx func start",
+    "test": "npm run build && mocha --require ts-node/register \"tests/**/*.test.ts\""
```
Comment (a1exwang) [inline adaptive-card-notification/bot/package.json:16]: maybe sync with latest template here? remove test and redundant dependencies? https://github.com/OfficeDev/TeamsFx/blob/c0092eafc0992890d60b1c1b95524eef76a9d88a/templates/bot/ts/notification-function-base/package.json#L13

File: adaptive-card-notification/bot/src/adaptiveCard.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export type AdaptiveCard = any;
+
+/**
+ * Build adaptive card payload with card template and data.
+ * @param getCardData Function to prepare your card data.
+ * @param cardTemplate The adaptive card template.
+ * @returns An adaptive card object.
+ */
+export function buildAdaptiveCard<TData>(
+  getCardData: () => TData,
```
Comment (a1exwang) [inline adaptive-card-notification/bot/src/adaptiveCard.ts:13]: the latest template changed callback to object.

File: adaptive-card-notification/.fx/configs/projectSettings.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+    "appName": "AdaptiveCardNotification",
+    "projectId": "dac6ca6b-e075-4ebc-9343-88b4c2f047e7",
```
Comment (a1exwang) [inline adaptive-card-notification/.fx/configs/projectSettings.json:3]: Sample app should have no project ID? For example the bot sample: https://github.com/OfficeDev/TeamsFx-Samples/blob/609c5aeadadcc1b27719e4f1154c8b5b45a09f48/hello-world-bot/.fx/configs/projectSettings.json#L1-L22

File: adaptive-card-notification/bot/.funcignore [hunk 5]
```diff
@@ -0,0 +1,13 @@
+*.js.map
```
[the remark this answers is not in the document]
Comment (XiaofuHuang) [reply adaptive-card-notification/bot/.funcignore]: Thx, sync the latest template.

File: adaptive-card-notification/.fx/configs/projectSettings.json (hunk 1 above)
Comment (XiaofuHuang) [reply adaptive-card-notification/.fx/configs/projectSettings.json]: Updated! thx!

**Review by a1exwang — APPROVED** [review]

File: adaptive-card-notification/.fx/configs/azure.parameters.dev.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "provisionParameters": {
+      "value": {
+        "resourceBaseName": "adaptivecadev22578f",
```
Comment (swatDong) [inline adaptive-card-notification/.fx/configs/azure.parameters.dev.json:7]: Hardcoded suffix?

File: adaptive-card-notification/bot/src/columnsetNotificationHttpTrigger.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { AzureFunction, Context, HttpRequest } from "@azure/functions";
+import { AdaptiveCards } from "@microsoft/adaptivecards-tools";
+import { CardData } from "./cardModels";
```
Comment (swatDong) [inline adaptive-card-notification/bot/src/columnsetNotificationHttpTrigger.ts:3]: `CardData` does not match the format of `*.data.json`

File: adaptive-card-notification/README.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+> 
+> We really appreciate your feedback! If you encounter any issue or error, please report issues to us following the [Supporting Guide](./../SUPPORT.md). Meanwhile you can make [recording](https://aka.ms/teamsfx-record) of your journey with our product, they really make the product better. Thank you!
+>  
+> This warning will be removed when the samples are ready for production.
+
+Adaptive Card Notification provides an easy way to send notification in Teams. The front end is built with Adaptive Cards to render notification details, the bot framework service is an Azure Bot Service handling search queries and communication between the server workload and the client and the backend is hosted in Azure Functions providing notification trigger and message handler.
+
+## Prerequisite
+- [NodeJS](https://nodejs.org/en/), fully tested on NodeJS 12, 14
+- An M365 account. If you do not have M365 account, apply one from [M365 developer program](https://developer.microsoft.com/en-us/microsoft-365/dev-program)
```
Comment (kimizhu) [inline adaptive-card-notification/README.md:13]: Is it needed? incoming webhook should not require a Teams app.

File: adaptive-card-notification/.vscode/tasks.json [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+                "start ngrok",
+                "prepare local environment",
+                "Start All"
+            ],
+            "dependsOrder": "sequence"
+        },
+        {
+            "label": "validate local prerequisites",
+            "type": "shell",
+            "command": "exit ${command:fx-extension.validate-local-prerequisites}",
```
Comment (swatDong) [inline adaptive-card-notification/.vscode/tasks.json:17]: @kuojianlu - when will we apply new F5 tasks to samples?

**Review by swatDong — APPROVED** [review]

**Pull request merged**